iml-api/src/graphql.rs

```diff
@@ -651,7 +692,7 @@ impl MutationRoot {
         interval: GraphQLDuration,
         use_barrier: Option<bool>,
     ) -> juniper::FieldResult<bool> {
-        sqlx::query!(
+        let maybe_id = sqlx::query!(
```

Comment:
What is returned when the we hit the ON CONFLICT DO NOTHING part?

If it's the same id, do we end up scheduling the same timer again?

Reply:
After adding a timer interval, I attempted to add the same interval a second time. Here are the results:

![image](https://user-images.githubusercontent.com/1100657/94027519-ec4bed80-fd88-11ea-9947-1263f50ed8e2.png)


```
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.456 DEBUG hyper::proto::h1::io: read 1900 bytes
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.456 DEBUG hyper::proto::h1::io: parsed 17 headers
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.456 DEBUG hyper::proto::h1::conn: incoming body is content-length (1189 bytes)
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.456 DEBUG hyper::proto::h1::conn: incoming body completed
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.457  INFO sqlx::query: /* SQLx ping */; rows: 0, elapsed: 390.074µs
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.459  INFO sqlx::query: INSERT INTO snapshot_interval ( …; rows: 0, elapsed: 1.747ms
iml_iml-api.1.s5hlsqf9at8q@adm.local    |
iml_iml-api.1.s5hlsqf9at8q@adm.local    | INSERT INTO
iml_iml-api.1.s5hlsqf9at8q@adm.local    |   snapshot_interval (
iml_iml-api.1.s5hlsqf9at8q@adm.local    |     filesystem_name,
iml_iml-api.1.s5hlsqf9at8q@adm.local    |     use_barrier,
iml_iml-api.1.s5hlsqf9at8q@adm.local    |     interval
iml_iml-api.1.s5hlsqf9at8q@adm.local    |   )
iml_iml-api.1.s5hlsqf9at8q@adm.local    | VALUES
iml_iml-api.1.s5hlsqf9at8q@adm.local    |   ($1, $2, $3) ON CONFLICT (filesystem_name, interval) DO NOTHING RETURNING id
iml_iml-api.1.s5hlsqf9at8q@adm.local    |
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.459 DEBUG iml_api: Some(V4(10.0.4.64:34968)) "POST /graphql HTTP/1.0" 200 "Some("https://localhost:7443/graphiql")" "Some("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36")" 3.097607ms
iml_iml-api.1.s5hlsqf9at8q@adm.local    | Sep 23 14:37:27.459 DEBUG hyper::proto::h1::io: flushed 157 bytes
```
Notice it did not send a PUT request to the timer. The timer logs did not log anything either. 

The next step is to check the timers. There should only be one snapshot timer, not two:
```
[root@iml-timer /]# systemctl list-timers
NEXT                         LEFT     LAST                         PASSED    UNIT                         ACTIVATES
Wed 2020-09-23 14:41:38 UTC  35s left Wed 2020-09-23 14:40:38 UTC  24s ago   iml-snapshot-12.timer        iml-snapsh
ot-12.service
Thu 2020-09-24 14:17:06 UTC  23h left Wed 2020-09-23 14:17:06 UTC  23min ago systemd-tmpfiles-clean.timer systemd-tm
pfiles-clean.service

2 timers listed.
```

This shows that a second timer was not added. In contrast, I will change the interval to a different value and I would expect that it would create a second timer:
```
[root@iml-timer /]# systemctl list-timers
NEXT                         LEFT          LAST                         PASSED    UNIT                         ACTIV
ATES
Wed 2020-09-23 14:42:38 UTC  13s left      Wed 2020-09-23 14:41:38 UTC  46s ago   iml-snapshot-12.timer        iml-s
napshot-12.service
Wed 2020-09-23 14:44:17 UTC  1min 52s left n/a                          n/a       iml-snapshot-15.timer        iml-s
napshot-15.service
Thu 2020-09-24 14:17:06 UTC  23h left      Wed 2020-09-23 14:17:06 UTC  25min ago systemd-tmpfiles-clean.timer syste
md-tmpfiles-clean.service

3 timers listed.
```

Reply:
Thanks